src/test/java/org/jabref/logic/citation/service/SearchCitationsRelationsServiceTest.java

```diff
@@ -0,0 +1,178 @@
+package org.jabref.logic.citation.service;
+
+import java.util.HashMap;
+import java.util.List;
+
+import org.jabref.logic.citation.repository.BibEntryRelationsRepositoryHelpersForTest;
+import org.jabref.logic.importer.fetcher.CitationFetcherHelpersForTest;
+import org.jabref.model.entry.BibEntry;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class SearchCitationsRelationsServiceTest {
+
+    @Nested
+    class CitationsTests {
+        @Test
+        void serviceShouldSearchForCitations() {
+            // GIVEN
+            var cited = new BibEntry();
+            var citationsToReturn = List.of(new BibEntry());
+            var repository = BibEntryRelationsRepositoryHelpersForTest.Mocks.from(
+                e -> citationsToReturn, null, null, null
+            );
+            var searchService = new SearchCitationsRelationsService(null, repository);
+
+            // WHEN
+            List<BibEntry> citations = searchService.searchCitations(cited, false);
+
+            // THEN
+            assertEquals(citationsToReturn, citations);
+        }
+
+        @Test
+        void serviceShouldForceCitationsUpdate() {
+            // GiVEN
+            var cited = new BibEntry();
+            var newCitations = new BibEntry();
+            var citationsToReturn = List.of(newCitations);
+            var citationsDatabase = new HashMap<BibEntry, List<BibEntry>>();
+            var fetcher = CitationFetcherHelpersForTest.Mocks.from(
+                entry -> {
+                    if (entry == cited) {
+                        return citationsToReturn;
+                    }
+                    return List.of();
+                },
+                null
```

Comment:
Not sure about the "null" here. But I think, it is OK for now.

We want to go away with nulls in JabRef. If we have it, we annotate with `jspecify`. But in tests, its ok.